### DISCUSSION TITLE
下载的CDLA中文版面分析模型评估与自己训练的bbox ap都很低

### DISCUSSION THREAD

**Discussion by Johnbarry2** [PaddlePaddle/PaddleOCR#13304] [Q&A]
问题描述：
下载中文版面分析模型评估bbox ap只有0.11，数据集格式转换用的项目中的x2coco.py,自己训练出来的bbox ap也只有0.35左右

 运行环境：
- OS:linux,两张4090
- Paddle:2.6.1
- PaddleOCR:2.7.3

![下载模型评估](https://github.com/PaddlePaddle/PaddleOCR/assets/174785530/ed481d80-f20b-4ea7-9bde-f67868baab8f)

picodet_lcnet_x1_0_layout配置文件：
_BASE_: [
  '../../../../runtime.yml',
  '../../_base_/picodet_esnet.yml',
  '../../_base_/optimizer_100e.yml',
  '../../_base_/picodet_640_reader.yml',
]

pretrain_weights: https://paddledet.bj.bcebos.com/models/pretrained/LCNet_x1_0_pretrained.pdparams
weights: output/picodet_lcnet_x1_0_layout/model_final
find_unused_parameters: True
use_ema: true
cycle_epoch: 10
snapshot_epoch: 1
epoch: 100

PicoDet:
  backbone: LCNet
  neck: CSPPAN
  head: PicoHead
  nms_cpu: True

LCNet:
  scale: 1.0
  feature_maps: [3, 4, 5]

metric: COCO
num_classes: 10

TrainDataset:
    name: COCODataSet
    image_dir: /home/cc009/zzj/paddle/code/PaddleDetection/CDLA_DATASET/train/
    anno_path: /home/cc009/zzj/paddle/code/PaddleDetection/CDLA_DATASET/train.json
    dataset_dir: /home/cc009/zzj/paddle/code/PaddleDetection/CDLA_DATASET/
    data_fields: ['image', 'gt_bbox', 'gt_class', 'is_crowd']

EvalDataset:
    name: COCODataSet
    image_dir: /home/cc009/zzj/paddle/code/PaddleDetection/CDLA_DATASET/val
    anno_path: /home/cc009/zzj/paddle/code/PaddleDetection/CDLA_DATASET/val.json
    dataset_dir: /home/cc009/zzj/paddle/code/PaddleDetection/CDLA_DATASET/

TestDataset:
  !ImageFolder
    anno_path: /home/cc009/zzj/paddle/code/PaddleDetection/CDLA_DATASET/val.json

worker_num: 8
eval_height: &eval_height 800
eval_width: &eval_width 608
eval_size: &eval_size [*eval_height, *eval_width]

TrainReader:
  sample_transforms:
  - Decode: {}
  - RandomCrop: {}
  - RandomFlip: {prob: 0.5}
  - RandomDistort: {}
  batch_transforms:
  - BatchRandomResize: {target_size: [[768, 576], [800, 608], [832, 640]], random_size: True, random_interp: True, keep_ratio: False}
  - NormalizeImage: {is_scale: true, mean: [0.485,0.456,0.406], std: [0.229, 0.224,0.225]}
  - Permute: {}
  batch_size: 32
  shuffle: true
  drop_last: true
  collate_batch: false

EvalReader:
  sample_transforms:
  - Decode: {}
  - Resize: {interp: 2, target_size: [800, 608], keep_ratio: False}
  - NormalizeImage: {is_scale: true, mean: [0.485,0.456,0.406], std: [0.229, 0.224,0.225]}
  - Permute: {}
  batch_transforms:
  - PadBatch: {pad_to_stride: 32}
  batch_size: 8
  shuffle: false

TestReader:
  inputs_def:
    image_shape: [1, 3, 800, 608]
  sample_transforms:
  - Decode: {}
  - Resize: {interp: 2, target_size: [800, 608], keep_ratio: False}
  - NormalizeImage: {is_scale: true, mean: [0.485,0.456,0.406], std: [0.229, 0.224,0.225]}
  - Permute: {}
  batch_transforms:
  - PadBatch: {pad_to_stride: 32}
  batch_size: 1
  shuffle: false

**Comment by GreatV** [comment]
配置文件用的是 `picodet_lcnet_x1_0_fgd_layout_cdla` 吗

**Comment by Johnbarry2** [comment]
> 配置文件用的是 `picodet_lcnet_x1_0_fgd_layout_cdla` 吗

用的是picodet_lcnet_x1_0_layout，修改了数据集路径和类别，训练的配置文件就是上文中的配置文件

**Comment by GreatV** [comment]
这个就是单纯的目标检测模型，你随便选个yolo系的模型，我估计map会很高。

**Comment by Johnbarry2** [comment]
> 这个就是单纯的目标检测模型，你随便选个yolo系的模型，我估计map会很高。

好的好的，感谢，我去试试，只是我看别人提的issue和我一样的预训练模型和数据，bbox ap能到80左右，我不知道是哪里出了问题，所以想请大佬解答一下

**Comment by Baicai0208** [comment]
我也有相同的问题，请问你成功了吗

**Comment by BotAndyGao** [comment]
第一，确定一下你的标注工具中layout的检测域的分类顺序，训练集、验证集的分类顺序必须一致，这个可以保证训练时验证的正确，bbox ap如果在你的标注集质量高的情况下，最少在92以上。
![categories](https://github.com/user-attachments/assets/1e4f820b-4403-46ae-9461-10acf198e1ad)
第二，确定你服务发布时的dict字典文件的顺序也要和你的训练集、验证集的区域分类的顺序一致，才能保证你发布的服务具有正确的推理结果。
equation
figure
figure_caption
footer
header
reference
table
table_caption
text
title
